packages/atlas/src/CommonProviders.tsx

```diff
@@ -13,12 +17,37 @@ export const CommonProviders: FC<PropsWithChildren> = ({ children }) => {
   return (
     <>
       <GlobalStyles />
-      <ApolloProvider client={apolloClient}>
-        <OperatorsContextProvider>
-          <AssetsManager />
-          <BrowserRouter>{children}</BrowserRouter>
-        </OperatorsContextProvider>
-      </ApolloProvider>
+      <OverlayManagerProvider>
+        <ApolloProvider client={apolloClient}>
+          <MaintenanceWrapper>
+            <OperatorsContextProvider>
+              <AssetsManager />
+              <BrowserRouter>{children}</BrowserRouter>
+            </OperatorsContextProvider>
+          </MaintenanceWrapper>
+        </ApolloProvider>
+      </OverlayManagerProvider>
+    </>
+  )
+}
+
+const MaintenanceWrapper: FC<PropsWithChildren> = ({ children }) => {
+  const { isKilled, wasKilledLastTime, error, loading } = useGetKillSwitch()
+
+  return (
+    <>
+      {isKilled || (error && wasKilledLastTime) || (loading && wasKilledLastTime) ? (
+        <>
+          <Maintenance />
+        </>
+      ) : (
+        children
+      )}
+      {(isKilled || (loading && wasKilledLastTime)) && !loading && !error && (
+        <BrowserRouter>
+          <AdminModal />
+        </BrowserRouter>
+      )}
```

Comment:
I don't like that `AdminModal` is conditionally rendered in two places. Maybe we just remove dependency on the router and place `AdminModal` directly next to `MaintenanceWrapper`?